src/openms/source/ANALYSIS/QUANTITATION/IsotopeLabelingMDVs.cpp

```diff
@@ -0,0 +1,342 @@
+          
+        for (size_t i = 0; i < normalized_feature.getSubordinates().size(); ++i)
+        {
+          if (measured_feature_max != 0.0) {
+            normalized_feature.getSubordinates().at(i).setIntensity((OpenMS::Peak2D::IntensityType)measured_feature_subordinates.at(i).getMetaValue(feature_name) / measured_feature_max);
+          }
+        }
+      }
+    }
+    
+    else if (mass_intensity_type == "norm_sum")
+    {
+      if (feature_name == "intensity")
+      {
+        OpenMS::Peak2D::IntensityType feature_peak_apex_intensity_sum = 0.0;
+        std::vector<OpenMS::Peak2D::IntensityType> intensities_vec;
+        for (auto it = measured_feature_subordinates.begin(); it != measured_feature_subordinates.end(); it++)
+        {
+          feature_peak_apex_intensity_sum += it->getIntensity();
+        }
+        
+        for (auto it = measured_feature_subordinates.begin(); it != measured_feature_subordinates.end(); it++)
+        {
+          normalized_feature.setMetaValue((it - measured_feature_subordinates.begin()), (it->getIntensity() / feature_peak_apex_intensity_sum));
+        }
+      }
+      
+      else
+      {
+        OpenMS::Peak2D::IntensityType feature_peak_apex_intensity_sum = 0.0;
+        std::vector<OpenMS::Peak2D::IntensityType> intensities_vec;
+        for (auto it = measured_feature_subordinates.begin(); it != measured_feature_subordinates.end(); it++)
+        {
+          feature_peak_apex_intensity_sum += (Peak2D::IntensityType)it->getMetaValue(feature_name);
+        }
+
+        for (size_t i = 0; i < normalized_feature.getSubordinates().size(); ++i)
+        {
+          if (feature_peak_apex_intensity_sum != 0.0)
+          {
+            normalized_feature.getSubordinates().at(i).setIntensity((OpenMS::Peak2D::IntensityType)measured_feature_subordinates.at(i).getMetaValue(feature_name) / feature_peak_apex_intensity_sum);
+          }
+        }
+      }
+    }
+  }
+
+  void IsotopeLabelingMDVs::calculateMDVs(
+    FeatureMap& measured_featureMap, FeatureMap& normalized_featureMap,
+    const String& mass_intensity_type, const String& feature_name)
+  {
+    for (FeatureMap::Iterator feature_it = measured_featureMap.begin(); feature_it != measured_featureMap.end(); feature_it++){
+      Feature normalized_feature;
+      calculateMDV(*feature_it, normalized_feature, mass_intensity_type, feature_name);
+      normalized_featureMap.push_back(normalized_feature);
+    }
+  }
+
+  template<typename T>
+  void IsotopeLabelingMDVs::inverseMatrix_(
+    std::vector<std::vector<T>>& correction_matrix,
+    std::vector<std::vector<T>>& correction_matrix_inversed)
+  {
+    uint16_t correction_matrix_n = correction_matrix.size() == correction_matrix[0].size() ? correction_matrix.size() : 0;
+
+    // 1- get the inverse
+    double **CM_temp, **CM_inv;
+    double temp;
+    int i, j, k;
+
+    CM_temp = (double **)malloc(correction_matrix_n * sizeof(double *));
```

Comment:
please don't use C-style manual memory management.
We use eigen for linear algebra in OpenMS so I would suggest to use that (which also comes with matrix inversion etc.)

Reply:
thanks for the recommendation, I have removed the function to rely solely on eigen for inversion